adal/tests/Test.ADAL.NET.UIAutomation/CoreMobileMSALTests.cs

```diff
@@ -0,0 +1,152 @@
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenTest(ITestController controller)
+        {
+            //Get User from Lab
+            var user = controller.GetUser(
+                new UserQueryParameters
+                {
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = false
+                });
+
+            //Clear Cache
+            controller.Tap("Cache");
+            controller.Tap("clearCache");
+
+            SetInputData(controller, UIAutomationAppV2, "User.Read");
+
+            PerformSignInFlowFlow(controller, user);
+
+            //Verify result. Test results are put into a label
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+        }
+
+        /// <summary>
+        /// Runs through the standard acquire token flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenSilentTest(ITestController controller)
+        {
+            //Get User from Lab
+            var user = controller.GetUser(
+                new UserQueryParameters
+                {
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = false
+                });
+
+            //Clear Cache
+            controller.Tap("Cache");
+            controller.Tap("clearCache");
+
+            SetInputData(controller, UIAutomationAppV2, "User.Read");
+
+            PerformSignInFlowFlow(controller, user);
+
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+
+            SetInputData(controller, UIAutomationAppV2, "User.Write");
+
+            controller.Tap("acquireTokenSilent");
+
+            //Verify result. Test results are put into a label
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+        }
+
+        /// <summary>
+        /// Runs through the standard acquire token interactive flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenADFSvXInteractiveMSALTest(ITestController controller, FederationProvider federationProvider, bool isFederated)
+        {
+            //Get User from Lab
+            var user = controller.GetUser(
+                new UserQueryParameters
+                {
+                    FederationProvider = federationProvider,
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = isFederated
+                });
+
+            //Clear Cache
+            controller.Tap("Cache");
+            controller.Tap("clearCache");
```

Comment:
could be consolidated into one and re-user elsewhere

Reply:
This has been consolidated into prepareForAuthentication()